packages/ui-components/src/features/Form/FormInput.tsx

```diff
@@ -30,11 +28,23 @@ export const FormInput = ({
 
   const requiredConfig = required ? { required: 'Required' } : {};
 
-  const verifyIsNonwhitespace = (value: string) => !!value.trim() || 'Must not be empty';
+  const verifyIsNonwhitespace = (value: string) => !!value.trim() || 'Required';
+
+  // prettier-ignore
   const validateConfig =
     typeof validate === 'function'
-      ? { validate: nonWhitespace ? { validate, verifyIsNonwhitespace } : { validate } }
-      : { validate: nonWhitespace ? { ...validate, verifyIsNonwhitespace } : { ...validate } };
+      ? {
+          validate:
+            required === 'non-whitespace'
```

Comment:
I think we can simplify this on step further and say that all required fields should check that the trimmed value is not empty (check first that the value is a string though). This is fairly standard behaviour I think, and it[ seems like a few people have been asking for it](https://github.com/react-hook-form/react-hook-form/issues/1650) from `react-hook-form` as well.

Reply:
Hm, if we trim to validate on all `required` fields, then we’d still need some way to opt password fields out of this behaviour. Otherwise, a password that’s long enough would be considered by React Hook Form to be too short.

That was my rationale for separating normal-required and trimmed-required; but I could try and figure out how to dynamically opt out of trim validation based on `type="password"`

Reply:
I guess we would then ask, will people ever have spaces in passwords? @tcaiger maybe you can chime in here?

Reply:
Our backend seems to accept it; as far as I can tell the only constraint we have on passwords is a minimum length. I don’t know how common for people to choose to include spaces, but I’ve definitely done it before

Speculating, but I think password managers/generators avoid it because some systems restrict passwords to alphanumerics + a handful of symbols

Reply:
Okay cool, well we can just do it for any inputs where type !== `password` and that should work.